packages/lib/delivery/src/notifications/broker/NotificationBroker.ts

```diff
@@ -55,10 +56,12 @@ export const _setupNotficationBroker = (
  */
 export const NotificationBroker = (
     notificationChannel: NotificationChannel[],
+    notificationType: NotificationType,
 ): INotificationBroker => {
     const channels = notificationChannel.map((channel) => {
         switch (channel.type) {
             case NotificationChannelType.EMAIL:
+                channel.config.notificationType = notificationType;
```

Comment:
Why do we assign the notificationType to the config ?

Reply:
It is added so that everytime notificationBroker( ) is called we can fetch which kind of email is to be send like : OTP or NEW MESSAGE. And then from here we pass notificationType to _setupNotficationBroker() which calls the send() method to send email. And in that send() we have that config and accordingly we can decide whether its a OTP email or a NEW MESSAGE EMAIL so we can fetch that email template.

Reply:
I think the channel.config.notificationType will never be referenced because its only accessible during the map call. Once the function has returned the Garbage collector will remove it

Reply:
map function will return then values to channels and then we are calling     

**return _setupNotficationBroker(channels);**

It should work

Reply:
I see NotificationType is the type of Mail that is about to be send (OTP, NEW_MESSAGE) right ? In that case please add a comment explaining what the difference between channel and type is. Because it is a bit confusing without knowing about it

Reply:
And please update the TS doc of NotificationBroker so it contains the right params

Reply:
Added comments & updated TS doc